File: app/src/main/java/com/ishow/noah/modules/sample/main/SampleManager.java
package com.ishow.noah.modules.sample.main;

import com.ishow.noah.R;
import com.ishow.noah.modules.sample.SampleStatusViewActivity;
import com.ishow.noah.modules.sample.dialog.normal.SampleBaseDialogActivity;
import com.ishow.noah.modules.sample.dialog.select.SampleSelectDialogAndPickerDialog;
import com.ishow.noah.modules.sample.edittextpro.SampleEditTextProActivity;
import com.ishow.noah.modules.sample.entries.Sample;
import com.ishow.noah.modules.sample.http.SampleHttpActivity;
import com.ishow.noah.modules.sample.imageloader.SampleImageLoaderActivity;
import com.ishow.noah.modules.sample.permission.SamplePermissionActivity;
import com.ishow.noah.modules.sample.photo.select.SampleSelectPhotoActivity;
import com.ishow.noah.modules.sample.pickview.SamplePickerActivity;
import com.ishow.noah.modules.sample.pulltorefresh.SamplePullToRefreshActivity;
import com.ishow.noah.modules.sample.recycle.animation.SampleAnimationRecycleViewActivity;
import com.ishow.noah.modules.sample.webview.loading.SampleLoadingWebViewActivity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by yuhaiyang on 2017/10/12.
 * 管理器
 */

public class SampleManager {


    public static List<Sample> getSampleCate() {
        List<Sample> list = new ArrayList<>();
        list.add(new Sample(R.string.sample_widget_list, SampleMainActivity.class));
        return list;
    }

    public static List<Sample> getSamples() {
        return getSamples(Sample.Cate.Widget);
    }


    @SuppressWarnings("WeakerAccess")
    public static List<Sample> getSamples(@Sample.Cate int cate) {
        switch (cate) {
            case Sample.Cate.Widget:
                return getWidgetSamples();
            default:
                return new ArrayList<>();
        }
    }

    private static List<Sample> getWidgetSamples() {
        List<Sample> list = new ArrayList<>();
        list.add(Sample.newInstance(R.string.sample_pick_view, SamplePickerActivity.class));
        list.add(Sample.newInstance(R.string.sample_select_photo, SampleSelectPhotoActivity.class));
        list.add(Sample.newInstance(R.string.sample_permission, SamplePermissionActivity.class));
        list.add(Sample.newInstance(R.string.sample_edittextpro, SampleEditTextProActivity.class));
        list.add(Sample.newInstance(R.string.sample_http, SampleHttpActivity.class));
        list.add(Sample.newInstance(R.string.sample_imageloader, SampleImageLoaderActivity.class));
        list.add(Sample.newInstance(R.string.sample_ani_recycle, SampleAnimationRecycleViewActivity.class));
        list.add(Sample.newInstance(R.string.sample_picker_and_select, SampleSelectDialogAndPickerDialog.class));
        list.add(Sample.newInstance(R.string.sample_loading_webview, SampleLoadingWebViewActivity.class));
        list.add(Sample.newInstance(R.string.sample_pull_to_refresh, SamplePullToRefreshActivity.class));
        list.add(Sample.newInstance(R.string.sample_base_dialog, SampleBaseDialogActivity.class));
        list.add(Sample.newInstance(R.string.sample_status_view, SampleStatusViewActivity.class));
        return list;
    }
}

File: app/src/main/java/com/ishow/noah/modules/sample/SampleManager.kt
package com.ishow.noah.modules.sample

import com.ishow.noah.modules.sample.detail.dashline.SampleDashLineFragment
import com.ishow.noah.modules.sample.detail.dialog.normal.SampleBaseDialogFragment
import com.ishow.noah.modules.sample.detail.dialog.select.SampleDataPickerDialogFragment
import com.ishow.noah.modules.sample.detail.edittextpro.SampleEditTextProFragment
import com.ishow.noah.modules.sample.detail.glide.SampleGlideCornerFragment
import com.ishow.noah.modules.sample.detail.log.SampleLogFragment
import com.ishow.noah.modules.sample.detail.permission.SamplePermissionFragment
import com.ishow.noah.modules.sample.detail.photo.select.SampleSelectPhotoFragment
import com.ishow.noah.modules.sample.detail.pickview.SamplePickerFragment
import com.ishow.noah.modules.sample.detail.recycle.adapter.SampleBindAdapterFragment
import com.ishow.noah.modules.sample.detail.utils.json.SampleJsonFragment
import com.ishow.noah.modules.sample.detail.utils.reflection.SampleReflectionFragment
import com.ishow.noah.modules.sample.detail.views.constraint.SampleConstraintLayoutFragment
import com.ishow.noah.modules.sample.detail.dialog.loading.SampleLoadingDialogFragment
import com.ishow.noah.modules.sample.detail.views.loading.SampleLoadingWebViewFragment
import com.ishow.noah.modules.sample.detail.views.pull2refresh.SamplePull2RefreshFragment
import com.ishow.noah.modules.sample.detail.views.statusview.SampleStatusViewFragment
import com.ishow.noah.modules.sample.entries.Sample
import java.util.*

/**
 * Created by yuhaiyang on 2017/10/12.
 * 管理器
 */

object SampleManager {


    val samples: MutableList<Sample>
        get() {
            val list = ArrayList<Sample>()
            list.add(Sample.instance("PickerView", SamplePickerFragment::class.java))
            list.add(Sample.instance("GlideCorner", SampleGlideCornerFragment::class.java))
            list.add(Sample.instance("StatusView", SampleStatusViewFragment::class.java))
            // Dialog
            list.add(Sample.instance("BaseDialog", SampleBaseDialogFragment::class.java))
            list.add(Sample.instance("LoadingDialog", SampleLoadingDialogFragment::class.java))
            list.add(Sample.instance("PickerData", SampleDataPickerDialogFragment::class.java))

            list.add(Sample.instance("EditTextPro", SampleEditTextProFragment::class.java))
            list.add(Sample.instance("LoadingWebView", SampleLoadingWebViewFragment::class.java))
            list.add(Sample.instance("DashLine", SampleDashLineFragment::class.java))
            list.add(Sample.instance("Permission", SamplePermissionFragment::class.java))
            list.add(Sample.instance("SelectPhoto", SampleSelectPhotoFragment::class.java))
            list.add(Sample.instance("Log", SampleLogFragment::class.java))
            list.add(Sample.instance("反射", SampleReflectionFragment::class.java))
            list.add(Sample.instance("Pull2Refresh", SamplePull2RefreshFragment::class.java))
            list.add(Sample.instance("Json", SampleJsonFragment::class.java))
            list.add(Sample.instance("ConstraintLayout", SampleConstraintLayoutFragment::class.java))
            list.add(Sample.instance("BindAdapter", SampleBindAdapterFragment::class.java))
            return list
        }
}
